doc/ Update Exchange Connector developer documentation

CrimsonJacket opened the pull request.

Nullably commented on content/developer/overview.mdx:
```diff
@@ -61,12 +69,12 @@ We encourage and welcome contributions from the community, subject to the guidel
 - The best way to create a connector that adheres to Hummingbot’s standard is by cloning the logic of existing connectors. We’ve done a lot of work to build our core connectors, so no need to reinvent the wheel.
```
I would add this as well

- Make sure every REST API call is done in asynchronous fashion.

Nullably commented on content/developer/task1.mdx:
```diff
@@ -63,165 +111,138 @@ Fetches or Subscribes to the order book deltas(diffs) for each trading pair. Add
+### `snapshot_message_from_exchange`
 
-### `active_bids`
+Converts json snapshot data from the exchangee into standard `OrderBookMessage` format.
```
typo, exchangee

Nullably commented on content/developer/task1.mdx:
```diff
@@ -63,165 +111,138 @@ Fetches or Subscribes to the order book deltas(diffs) for each trading pair. Add
 
-**Input Parameter:** None <br/>
-**Expected Output(s):** `Dict[Decimal, Dict[str, Dict[str, any]]]`
+### `snapshot_message_from_db`
```
I don't think we should mention this or anything that is not typically required.

Nullably commented on content/developer/task1.mdx:
```diff
@@ -63,165 +111,138 @@ Fetches or Subscribes to the order book deltas(diffs) for each trading pair. Add
 
-**Input Parameter:** `object`: message <br/>
-**Expected Output(s):** `Tuple[List[OrderBookRow],List[OrderBookRow]]`
+### `diff_message_from_db`
```
same, no need to have this in the doc.

Nullably commented on content/developer/task1.mdx:
```diff
@@ -63,165 +111,138 @@ Fetches or Subscribes to the order book deltas(diffs) for each trading pair. Add
+### `trade_message_from_exchange`
 
-### `c_convert_snapshot_message_to_np_arrays`
+Converts json trade data from the exchangee into standard `OrderBookMessage` format.
```
typo

Nullably commented on content/developer/task3.mdx:
```diff
@@ -1,444 +1,560 @@
+This function is responsible for executing the API request to place the order on the exchange. It does the following:
 
-The table below details the **required** functions to implement
+- Verifies that the orders would be legal based on the trading rules pulled from the exchange.
```
~~Verifies that the orders would be legal based on the trading rules pulled from the exchange.~~
Verifies that the order satisfies exchange trading rules.

Nullably reviewed (changes requested):
Well written documentation. 
Thanks a lot @CrimsonJacket  for taking the time to do this.
Some minor requested changes.

CrimsonJacket commented:
@Nullably Addressed your comments and made additional revisions. If it is sufficient, lets push it to `master` sooner rather than later. Seems like there's some feedback on the `#dev-*` channels wrt to updating our developer docs.

Nullably reviewed (approved):
LGTM

dennisocana commented:
Resolved a conflict due to a recent change in `Callout.js` and merged the PR.
